src/iqstat.cpp

```diff
@@ -357,3 +227,116 @@ void usage() {
     }
     printReleaseInfo( "iqstat" );
 }
+
+auto show_RuleExec( char *userName,
+                    const char *ruleName,
+                    int allFlag,
+                    bool brief ) -> int
+{
+    int status;
+    genQueryInp_t genQueryInp{};
+    genQueryOut_t *genQueryOut{};
+
+    output_format_parameters output_format;
+
+    auto num_cols_selected = (brief ? output_format.brief_format_len
+                                    : output_format.long_format_len);
+
+    for (int i = 0; i < num_cols_selected; i++) {
+        addInxIval(&genQueryInp.selectInp, output_format.columns[i], 0);
+    }
+
+    char v1[BIG_STR];
+    char v2[BIG_STR];
+
+    if (!allFlag) {
+        snprintf( v1, BIG_STR, "='%s'", userName );
+        addInxVal(&genQueryInp.sqlCondInp, COL_RULE_EXEC_USER_NAME, v1);
+    }
+
+    std::string diagnostic;
+    if (ruleName != NULL && *ruleName != '\0') {
+        diagnostic = (boost::format( " (and matching key '%s')" ) % ruleName).str();
+        snprintf( v2, BIG_STR, "='%s'", ruleName );
+        addInxVal(&genQueryInp.sqlCondInp, COL_RULE_EXEC_ID, v2);
+    }
+
+    genQueryInp.maxRows = 50;
+
+    // idempotent version of clearGenQueryOut.
+    auto clear_gq_out = [&o = genQueryOut]{ clearGenQueryOut(o);
+                                            o->attriCnt = 0; };
+    // RAII for both GenQuery variables.
+    auto clean_gq_in_out = [&, &i = genQueryInp, &o = genQueryOut]{ clearGenQueryInp(&i);
+                                                                    clear_gq_out();
+                                                                    free(o); o = nullptr; };
+    // Implement RAII:
+    const auto clean_gq = irods::at_scope_exit{[&] { clean_gq_in_out(); }};
+
+    status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+
+    if ( status == CAT_NO_ROWS_FOUND ) {
```

Comment:
Should we handle other `status < 0` case(s)?

Reply:
Not sure the best approach.  Should the error stack be printed in that case.  Should the status < 0 be converted to an irods excption and then thrown for the main( ) function to catch ?

Reply:
I was just commenting on the fact that there does not seem to be any error handling here and it may be worth just printing a message and getting out if `status < 0` (and is `!= CAT_NO_ROWS_FOUND`, of course)

Reply:
Seems good.

Reply:
Ok, think I  took care of this one. Please review change and lmk.